src/cargo/ops/cargo_compile.rs

```diff
@@ -116,12 +116,19 @@ pub enum CompileFilter<'a> {
 
 pub fn compile<'a>(ws: &Workspace<'a>, options: &CompileOptions<'a>)
                    -> CargoResult<ops::Compilation<'a>> {
+    compile_with_exec(ws, options, &mut DefaultExecutor)
+}
+
+pub fn compile_with_exec<'a, E: Executor>(ws: &Workspace<'a>,
+                                          options: &CompileOptions<'a>, 
+                                          exec: &mut E)
```

Comment:
Why is monomorphisation a problem? I'd have thought using generics is more efficient here (the common case is to just have a single Executor) and adding an extra param doesn't seem too much of a burden.